pkg/endpoint/endpoint.go

```diff
@@ -1949,6 +1954,8 @@ func (e *Endpoint) identityLabelsChanged(ctx context.Context, myChangeRev int) (
 		Debug("Assigned new identity to endpoint")
 
 	e.SetIdentity(allocatedIdentity, false)
+	// Since we has re-locked the endpoint
+	e.lastUpdatedTime[allocatedIdentity.StringID()] = time.Now()
```

Comment:
A bit confused why is this keyed on identity? Should it be using endpoint ID as key?

Reply:
as my understanding, this metric  meature the lantency like  cni/user(`cilium endpoint...`) interact with cilium-agent, and cilium-agent observed the result from k8s server. so this is why using an map record timestamp, because any user in node could change the endpoint at the same time,maybe i'm wrong:), and this also answer question 2,  use identity as key to ensure unique. can't use one value , otherwise the calculate result will always be last change. as @sayboras above description.

Reply:
hmm, I don't think that's what this is for. The metric itself captures the roundtrip delay of a CEP traveling from local node to apiserver and back. Each CEP should have its own value, not identity. The key should be CiliumEndpointID.

Reply:
:), thx! i take a try

Reply:
> hmm, I don't think that's what this is for. The metric itself captures the roundtrip delay of a CEP traveling from local node to apiserver and back. Each CEP should have its own value, not identity. The key should be CiliumEndpointID.

:), i think you mean CiliumEndpointUID?

Reply:
I meant this: https://github.com/cilium/cilium/blob/master/pkg/k8s/apis/cilium.io/v2/types.go#L47